Core/Utilities/ExecutorDispatcher.py

```diff
@@ -428,421 +431,416 @@ def __freezeTask( self, taskId, errMsg, eType = False, freezeTime = 60 ):
-    result = self.__msgTaskToExecutor( taskId, eId, eType )
-    if not result[ 'OK' ]:
-      self.__queues.pushTask( eType, taskId, ahead = True )
-      self.__states.removeTask( taskId )
+    self.__log.verbose("Sending task %s to %s=%s" % (taskId, eType, eId))
+    self.__states.addTask(eId, taskId)
+    result = self.__msgTaskToExecutor(taskId, eId, eType)
+    if not result['OK']:
+      self.__queues.pushTask(eType, taskId, ahead=True)
+      self.__states.removeTask(taskId)
       return result
-    return S_OK( taskId )
+    return S_OK(taskId)
 
-  def __msgTaskToExecutor( self, taskId, eId, eType ):
+  def __msgTaskToExecutor(self, taskId, eId, eType):
     try:
-      self.__tasks[ taskId ].sendTime = time.time()
+      self.__tasks[taskId].sendTime = time.time()
     except KeyError:
-      return S_ERROR( "Task %s has been deleted" % taskId )
+      return S_ERROR("Task %s has been deleted" % taskId)
     try:
-      result = self.__cbHolder.cbSendTask( taskId, self.__tasks[ taskId ].taskObj, eId, eType )
-    except:
-      self.__log.exception( "Exception while sending task to executor" )
-      return S_ERROR( "Exception while sending task to executor" )
-    if isReturnStructure( result ):
+      result = self.__cbHolder.cbSendTask(taskId, self.__tasks[taskId].taskObj, eId, eType)
+    except BaseException:
+      self.__log.exception("Exception while sending task to executor")
+      return S_ERROR("Exception while sending task to executor")
+    if isReturnStructure(result):
       return result
-    else:
-      errMsg = "Send task callback did not send back an S_OK/S_ERROR structure"
-      self.__log.fatal( errMsg )
-      return S_ERROR( "Send task callback did not send back an S_OK/S_ERROR structure" )
-    #Seems an executor problem
-    self.__log.verbose( "Disconnecting executor" )
-    self.removeExecutor( eId )
-    return S_ERROR( "Exception while sending task to executor" )
+    errMsg = "Send task callback did not send back an S_OK/S_ERROR structure"
+    self.__log.fatal(errMsg)
+    return S_ERROR(errMsg)
+
 
 if __name__ == "__main__":
   def testExecState():
     execState = ExecutorState()
-    execState.addExecutor( 1, "type1", 2 )
-    print execState.freeSlots( 1 ) == 2
-    print execState.addTask( 1, "t1" ) == 1
-    print execState.addTask( 1, "t1" ) == 1
-    print execState.addTask( 1, "t2" ) == 2
-    print execState.freeSlots( 1 ) == 0
-    print execState.full( 1 ) == True
-    print execState.removeTask( "t1" ) == 1
-    print execState.freeSlots( 1 ) == 1
-    print execState.getFreeExecutors( "type1" ) == {1:1}
-    print execState.getTasksForExecutor( 1 ) == [ "t2" ]
-    print execState.removeExecutor( 1 )
+    execState.addExecutor(1, "type1", 2)
+    print execState.freeSlots(1) == 2
+    print execState.addTask(1, "t1") == 1
+    print execState.addTask(1, "t1") == 1
+    print execState.addTask(1, "t2") == 2
+    print execState.freeSlots(1) == 0
+    print execState.full(1)
```

Comment:
This is not quite the same as the previous statement. The output of `execState.full(1)` is not necessarily a boolean, yet the previous statement would display a boolean

```
In [6]: 1 == True
Out[6]: True
```

Reply:
I didn't change these lines, autopep8 did.